oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/IDManager.java

```diff
@@ -252,6 +255,21 @@ public static EndpointRelationDefine analysisRelationId(String entityId) {
         }
     }
 
+    /**
+     * Process ID related functions.
+     */
+    public static class ProcessID {
+        /**
+         * @param instanceId built by {@link ServiceInstanceID#buildId(String, String)}
+         * @param name process name
+         * @return process id
+         */
+        public static String buildId(String instanceId, String name) {
+            return Hashing.sha256().newHasher().putString(String.format("%s_%s",
```

Comment:
Just curious, why do we choose a different encoding machanism `sha256` than the existing one `base64`?

Reply:
Yes, we need to keep the length fixed. The instance ID is closing to 512 which is the limitation for ElasticSearch/OpenSearch. But process belongs to an instance, traditional base64 is not going to work well.